### PR TITLE
Throw error if `trustedIssuer.id` is undefined.

### PR DESCRIPTION
**Pull request by JSAssassin**

### CODE REVIEW

File: lib/VerifiableCredentialStore.js [hunk 1]
```diff
@@ -167,7 +167,12 @@ export default class VerifiableCredentialStore {
 
       // build query to find all VCs that match any combination of type+issuer
       const query = [];
-      const issuers = trustedIssuers.map(({id}) => id);
+      const issuers = trustedIssuers.map(({id}) => {
+        if(!id) {
+          throw new Error('"id" must be a string.');
```
Comment (dlongley) [inline lib/VerifiableCredentialStore.js:172]: I have two concerns here:

1. What is the UX when this error is thrown?
2. We need to handle cases where no `id` is set in the future, but other attributes are provided (such as a nested VC query indicating a chain of trust use case, e.g., where any issuer with a VC that says they are an accredited university  will be trusted). We don't have to handle this now, but the error should probably indicate that the query is unsupported vs. it being invalid.

Comment (JSAssassin) [reply lib/VerifiableCredentialStore.js]: 1. So, the error on the UX currently is the same as when a non-matching trustedIssuer is sent in query - It says `Sorry, we can't seem to find any credentials.` 
2. I've updated the error message to indicate that a trustedIssuer without an id is unsupported.

File: lib/VerifiableCredentialStore.js [hunk 2]
```diff
@@ -167,7 +167,12 @@ export default class VerifiableCredentialStore {
 
       // build query to find all VCs that match any combination of type+issuer
       const query = [];
-      const issuers = trustedIssuers.map(({id}) => id);
+      const issuers = trustedIssuers.map(({id}) => {
+        if(!id) {
+          throw new Error('trustedIssuer without an "id" is unsupported.');
```
Comment (dlongley) [inline lib/VerifiableCredentialStore.js:172]:
```suggestion
          const error = new Error('trustedIssuer without an "id" is unsupported.');
          error.name = 'NotSupportedError';
          throw error;
```

File: test/web/10-VerifiableCredentialStore.js [hunk 3]
```diff
@@ -139,6 +139,34 @@ describe('VerifiableCredentialStore', () => {
...
+    try {
+      credentials = await vcStore.match({
+        query: queryWithTrustedIssuerWithoutId
+      });
+    } catch(e) {
+      err = e;
+    }
+
+    should.exist(err);
+    err.message.should.equal('trustedIssuer without an "id" is unsupported.');
```
Comment (dlongley) [inline test/web/10-VerifiableCredentialStore.js:166]: We don't want any tests to depend on error message english text, but rather on error `name`.

```suggestion
    err.name.should.equal('NotSupportedError');
```

**Review by dlongley — CHANGES_REQUESTED** [review]

File: lib/VerifiableCredentialStore.js (hunk 2 above)
Comment (JSAssassin) [reply lib/VerifiableCredentialStore.js]: Updated here 7eebda6

File: test/web/10-VerifiableCredentialStore.js (hunk 3 above)
Comment (JSAssassin) [reply test/web/10-VerifiableCredentialStore.js]: Updated 7eebda6

File: lib/VerifiableCredentialStore.js [hunk 4]
```diff
@@ -167,7 +167,15 @@ export default class VerifiableCredentialStore {
 
       // build query to find all VCs that match any combination of type+issuer
       const query = [];
-      const issuers = trustedIssuers.map(({id}) => id);
+      const issuers = trustedIssuers.map(({id}) => {
+        if(!id) {
+          const error =
+            new Error('trustedIssuer without an "id" is unsupported.');
```
Comment (dlongley) [inline lib/VerifiableCredentialStore.js:173]: Break on parenthesis when possible not around equals assignment.

```suggestion
          const error = new Error(
            'trustedIssuer without an "id" is unsupported.');
```

**Review by dlongley — APPROVED** [review]
Thanks! See one style nit comment.

File: lib/VerifiableCredentialStore.js (hunk 4 above)
Comment (JSAssassin) [reply lib/VerifiableCredentialStore.js]: Fixed 41a1f07

**Review by dmitrizagidulin — APPROVED** [review]
:+1:

**Pull request merged**